scripts/deploy

```diff
@@ -0,0 +1,122 @@
+# then delete only hpcc/data
+elif [[ "$pods" == *"Unable"* ]] || [[ "$pods" == *"No resources found in default namespace"* ]];then
+  # force rm data/config.json in hpcc only
+  echo "Forcibly delete hpcc/data only"
+  rm -vrf hpcc/data;cd hpcc; rm *.tfstate*;rm .terraform.lock.hcl ;sudo rm -r .terraform;cd ..
+fi
+#------------------------------------------------------------------------
+# See if $dir/data/config.json exists, where $dir is vnet or aks or storage.
+# If $dir/data/config.json does exist then from the file get
+# the resource group name and then check to see if resource group exists.
+# if it doesn't exists then delete $dir/data/config.json
+#------------------------------------------------------------------------
+for dir in aks vnet storage;do
+  if [ -e "$dir/data/config.json" ];then
+    # Get resource group name from file
+    rg=`$thisdir/get_rg_from_file $dir/data/config.json`
+    rgexist=`az group exists --name $rg`
+    if [ "$rgexist" == "false" ]; then
+       echo "In $0. $dir resource group, \"$rg\" does not exists. So deleting vnet/data and any tfstate files id $dir."
+       rm -vrf $dir/data
+       cd $dir; rm *.tfstate*;rm .terraform.lock.hcl ;sudo rm -r .terraform;cd ..
+  else
+       echo "In $0. $dir resource group, \"$rg\" DOES exists. So NOT deleting vnet/data."
+    fi
+  fi
+done
+#------------------------------------------------------------------------
+
+cd $name; # cd into vnet or storage or aks or hpcc
+
+echo "Copy metadata.tf in root directory to this directory."
+cp -v ../scripts/needed-auto-tfvars-files/metadata.tf .
+
+# put the root directory's lite.auto.tfvars (either all or part) in either aks or hpcc
+# directory.
+if [ -e "../lite.auto.tfvars" ];then 
+  # Check if there has been a change since last apply.
+  if [ -e "/tmp/${name}.lite.auto.tfvars" ];then
+    tfvarsdiff=`diff /tmp/${name}.lite.auto.tfvars ../lite.auto.tfvars`
+  else
+    tfvarsdiff=""
+  fi
+else
+  echo "In $0. The root directory does not have a file called 'lite.aute.tfvars'. It must. EXITING";exit 1;
+fi
+if [ "$name" == "hpcc" ];then
+    echo "Coping root's lite.auto.tfvars to /tmp and $name directory."
+    cp -v ../lite.auto.tfvars /tmp/${name}.lite.auto.tfvars
+    cp -v ../lite.auto.tfvars .
+    cp -v ../lite-variables.tf .
+elif [ "$name" == "aks" ] || [ "$name" == "vnet" ] || [ "$name" == "storage" ];then
+    egrep "^aks_" ../lite.auto.tfvars > /tmp/${name}.lite.auto.tfvars
+    egrep "^aks_" ../lite.auto.tfvars > lite.auto.tfvars
+    ../scripts/extract-aks-variables ../lite-variables.tf > lite-variables.tf
+    if [ "$name" == "storage" ];then
+       cp -v $thisdir/needed-auto-tfvars-files/storage/storage.auto.tfvars.example ./storage.auto.tfvars
+    fi
+fi
+#------------------------------------------------------------------------
+if [ ! -d "$HOME/tflogs" ];then
+  mkdir $HOME/tflogs
+fi
+
+plan=`$thisdir/mkplan deploy_${name}.plan`
```

Comment:
The use of backticks in bash scripts has been deprecated for some time now, in favor of using $() syntax.  See https://stackoverflow.com/questions/4708549/what-is-the-difference-between-command-and-command-in-shell-programming (and linked articles within that SO page) for more information.

Please use $() syntax here and in all other bash scripts.

Reply:
TLH: Replaced `...` with $(...)

Reply:
fixed